test/libsolidity/syntaxTests/specialFunctions/encodeCall_fail_funType.sol

```diff
@@ -0,0 +1,72 @@
+interface I {
+	function fExternal(uint256 p, string memory t) external;
+}
+
+contract Other {
+	function fExternal(uint) external pure {}
+	function fPublic(uint) public pure {}
+	function fInternal(uint) internal pure {}
+}
+
+library L {
+	function fExternal(uint256 p, string memory t) external {}
+	function fInternal(uint256 p, string memory t) internal {}
+}
+
+contract Base {
+	function baseFunction(uint) internal pure {}
+	function baseFunctionPublic(uint) public pure {}
+}
+
+contract C is Base {
+	using L for uint256;
+
+	function f(int a) public {}
+	function f2(int a, string memory b) public {}
+	function f3(int a, int b) public {}
+	function f4() public {}
+	function fInternal(uint256 p, string memory t) internal {}
+
+	function failFunctionPtrMissing() public returns(bytes memory) {
+		return abi.encodeCall(1, this.f);
+	}
+	function failFunctionPtrWrongType() public returns(bytes memory) {
+		return abi.encodeCall(abi.encodeCall, (1, 2, 3, "test"));
+	}
+	function failFunctionInternal() public returns(bytes memory) {
+		return abi.encodeCall(fInternal, (1, "123"));
+	}
+	function failFunctionInternalFromVariable() public returns(bytes memory) {
+		function(uint256, string memory) internal localFunctionPointer = fInternal;
+		return abi.encodeCall(localFunctionPointer, (1, "123"));
```

Comment:
The error here is:
```
Error: Function must be "public" or "external". Did you forget to prefix "this."?
```
I think we should not add this part about `this` when it's a pointer stored in a variable.

Reply:
Oh interesting that it still has "hasDeclaration" set to true...

Reply:
Ah sorry, I think the error you quoted is from line 34, i.e. `abi.encodeCall(abi.encodeCall`